app/models/registration.rb

```diff
@@ -445,4 +455,44 @@ def ensure_waitlist_eligibility!
   def serializable_hash(options = nil)
     super(DEFAULT_SERIALIZE_OPTIONS.merge(options || {}))
   end
+
+  def auto_accept
+    return log_error('Auto-accept is not enabled for this competition.') unless competition.auto_accept_registrations
+    return log_error('Can only auto-accept pending registrations or first position on waiting list') unless
+      competing_status_pending? || (competing_status_waiting_list? && waiting_list_position == 1)
+    return log_error("Competition has reached auto_accept_disable_threshold of #{competition.auto_accept_disable_threshold} registrations") if
+      competition.auto_accept_threshold_reached?
+    return log_error('Competitor still has outstanding registration fees') if outstanding_entry_fees > 0
+    return log_error('Cant auto-accept while registration is not open') if !competition.registration_currently_open?
+
+    if competition.registration_full_and_accepted? && competing_status_pending?
+      update_lanes!(
+        { user_id: user_id, competing: { status: Registrations::Helper::STATUS_WAITING_LIST } }.with_indifferent_access,
+        'Auto-accept',
+      )
+    else
+      update_lanes!(
+        { user_id: user_id, competing: { status: Registrations::Helper::STATUS_ACCEPTED } }.with_indifferent_access,
+        'Auto-accept',
+      )
+    end
+  rescue ActiveRecord::RecordInvalid => e
+    log_error("Auto accept error for registration #{id}: #{e}")
+  end
+
+  private def log_error(error)
+    Rails.logger.error(error)
+    false
+  end
+
+  private def waiting_list_auto_accept_check
```

Comment:
Why is the registration that is currently being accepted triggering auto-accept on the first registration of the waiting list?

Reply:
- [ ] Change this to `competing_status_previously_was`
- [ ] Try some `attempt_to_fill_opened_spot` to clarify what this method is trying to do